Lib/ufo2ft/otfPostProcessor.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import print_function, division, absolute_import, unicode_literals
+
+import tempfile
+
+from fontTools.ttLib import TTFont
+
+
+class OTFPostProcessor(object):
+    """Does some post-processing operations on a compiled OpenType font, using
+    info from the source UFO where necessary.
+    """
+
+    def __init__(self, otf, ufo):
+        self.ufo = ufo
+        tmp_file = tempfile.NamedTemporaryFile()
+        tmp_path = tmp_file.name
+        otf.save(tmp_path)
+        self.otf = TTFont(tmp_path)
+
+    def process(self, useProductionNames=True):
+        if useProductionNames:
+            self._rename_glyphs_from_ufo()
+        return self.otf
+
+    def _rename_glyphs_from_ufo(self):
```

Comment:
You're right indeed.

On Mon, May 2, 2016 at 8:09 PM, Cosimo Lupo <notifications@github.com>
wrote:

> In Lib/ufo2ft/otfPostProcessor.py
> <https://github.com/googlei18n/ufo2ft/pull/38#discussion_r61777605>:
>
> > +    info from the source UFO where necessary.
> > +    """
> > +
> > +    def __init__(self, otf, ufo):
> > +        self.ufo = ufo
> > +        tmp_file = tempfile.NamedTemporaryFile()
> > +        tmp_path = tmp_file.name
> > +        otf.save(tmp_path)
> > +        self.otf = TTFont(tmp_path)
> > +
> > +    def process(self, useProductionNames=True):
> > +        if useProductionNames:
> > +            self._rename_glyphs_from_ufo()
> > +        return self.otf
> > +
> > +    def _rename_glyphs_from_ufo(self):
>
> You're right, renaming works even without the "post table trick". At
> least, it seems it's sufficient to reload the TTFont just before doing
> getGlyphOrder (thus, decompiling the post table) and then immediately
> after do a setGlyphOrder, just like @jamesgk <https://github.com/jamesgk>
> does here. By the time the post table is compiled, the new glyphOrder is
> used.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/googlei18n/ufo2ft/pull/38/files/754ff3bf3d9ddddecc07af1c4af0d966983dcff8..55897b981ba133967c5a1276d3d48ea26b634652#r61777605>
>



-- 
behdad
http://behdad.org/